python/test/unit/language/test_core.py

```diff
@@ -1346,7 +1348,7 @@ def test_atomic_rmw(op, dtype_x_str, mode, sem, device):
         if dtype_x_str == 'float16':
             pytest.skip("Only test atomic float16 ops on GPU")
     else:
-        check_cuda_only(device)
+        check_cuda_or_hip(device)
```

Comment:
For cases the `torch.cuda` is only used for `torch.cuda.get_device_capability()` to determine if the test case should be skipped, can we guard `capability = torch.cuda.get_device_capability()` with the early skip under `is_cuda()` instead of exiting the `check_cuda_only`?

```
  if is_cuda():
    capability = torch.cuda.get_device_capability()
    if capability[0] < 7:
        if dtype_x_str == 'float16':
            pytest.skip("Only test atomic float16 ops on devices with sm >= 70")
```

Reply:
@ThomasRaoux can speak to this.

Reply:
my understanding is the torch hip driver can "behave as cuda" and still run through those tests right now. It is on AMD's list to clean this up at some point. Hopefully this is not too confusing for downstream backends.

Reply:
ah I see, we don't officially support anything before sm7 so I think the whole thing can be removed and we can just keep the check for interpreter. Feel free to send a PR.

Reply:
Created https://github.com/triton-lang/triton/pull/3955, thanks!